direct/core/src/test/java/cz/o2/proxima/commitlog/CommitLogReadersTest.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright 2017-2020 O2 Czech Republic, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.o2.proxima.commitlog;
+
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import cz.o2.proxima.direct.commitlog.CommitLogReader;
+import cz.o2.proxima.direct.commitlog.CommitLogReaders;
+import cz.o2.proxima.direct.commitlog.LogObserver;
+import cz.o2.proxima.storage.Partition;
+import cz.o2.proxima.storage.ThroughputLimiter;
+import cz.o2.proxima.storage.commitlog.Position;
+import java.time.Duration;
+import java.util.Arrays;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class CommitLogReadersTest {
+  private static class NoOpThroughputLimiter implements ThroughputLimiter {
+    @Override
+    public Duration getPauseTime(ThroughputLimiter.Context context) {
+      return Duration.ZERO;
+    }
+
+    @Override
+    public void close() {
+      // no-op
+    }
+  }
+
+  @Test
```

Comment:
Minor: can we add `@Test(expected = Test.None.class /* no exception expected */)` here? Currently this violates `squid:S2699` and this declaration makes it clear that the purpose of the test is to validate that no exception is thrown. See https://stackoverflow.com/questions/17731234/how-to-test-that-no-exception-is-thrown (second comment).

Reply:
Strongly disagree, as this is redundant (`expected` defaults to this value). Sonar is full of weird rules that shouldn't be blindly followed, this is one of them.